src/python/dxpy/workflow_builder.py

```diff
@@ -383,28 +421,37 @@ def _build_regular_workflow(json_spec, keep_open=False):
     return workflow_id
 
 
-def _get_validated_enabled_regions(json_spec, from_command_line):
+def _get_validated_enabled_regions(json_spec, args):
     """
     Returns a set of regions (region names) in which the global workflow
     should be enabled. Also validates and synchronizes the regions
     passed via CLI argument and in the regionalOptions field.
     """
+    # First determine in which regions the global workflow needs to be available
     enabled_regions = dxpy.executable_builder.get_enabled_regions('globalworkflow',
                                                                   json_spec,
-                                                                  from_command_line,
+                                                                  args.region,
                                                                   WorkflowBuilderException)
     if not enabled_regions:
         enabled_regions = []
-        if not dxpy.WORKSPACE_ID:
+        if not dxpy.PROJECT_CONTEXT_ID:
             msg = "A context project must be selected to enable a workflow in the project's region."
             msg += " You can use 'dx select' to select a project. Otherwise you can use --region option"
             msg += " to select a region in which the workflow should be enabled"
             raise(WorkflowBuilderException(msg))
-        region = dxpy.api.project_describe(dxpy.WORKSPACE_ID,
+        current_selected_region = dxpy.api.project_describe(dxpy.PROJECT_CONTEXT_ID,
                                            input_params={"fields": {"region": True}})["region"]
 
-        enabled_regions.append(region)
+        enabled_regions.append(current_selected_region)
 
+    # Get billable regions
+    enabled_regions = set(enabled_regions)
+    billable_regions = dxpy.executable_builder.get_permitted_regions(args.bill_to, WorkflowBuilderException)
+    enabled_regions.intersection_update(billable_regions)
+    
+    # Verify all the stages are also enabled in these regions
+    enabled_regions = _assert_executable_regions_match(enabled_regions, json_spec)
+    
     if not enabled_regions:
         raise AssertionError("This workflow should be enabled in at least one region")
```

Comment:
The enabled region(s) will be the union of the following:
- regionalOptions of source workflow spec(if exists)
- `--region` from CLI (if given)
- the permitted regions of billTo
- the dependent executables listed in the source workflow's description.